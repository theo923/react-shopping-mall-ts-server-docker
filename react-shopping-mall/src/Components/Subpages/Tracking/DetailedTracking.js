"use strict";
exports.__esModule = true;
exports.DetailedTracking = void 0;
var React = require("react");
var CheckoutInfo_1 = require("../ShoppingCart/Checkout/CheckoutInfo");
var CheckoutItem_1 = require("../ShoppingCart/Checkout/CheckoutItem");
var DetailedTracking = function (_a) {
    var order = _a.order, idx = _a.idx;
    var reducer = function (accumulator, currentObject) { return accumulator + (currentObject.quan * currentObject.price); };
    var orderedItem = order[0];
    var orderedDeliveryInfo = order[1];
    return (React.createElement("div", null,
        React.createElement("div", null,
            React.createElement("div", { className: "container glass text-neutral-content mb-6" },
                React.createElement("figure", { className: "px-6 relative" },
                    React.createElement("div", { style: { width: '266.67px', height: '150px' }, className: "float-right" },
                        React.createElement("img", { src: orderedItem[0].url, alt: orderedItem[0].name }))),
                React.createElement("div", { className: "card-body max-h-full max-w-full relative" },
                    React.createElement("h2", { className: "card-title" },
                        "Order #",
                        idx),
                    React.createElement("div", null, orderedItem.map(function (item, idx) { return React.createElement("div", { key: idx, className: 'italic' },
                        item.name,
                        " $",
                        item.quan * item.price,
                        " @ ",
                        item.quan,
                        " "); }))))),
        React.createElement("table", { className: "table w-full" },
            React.createElement("thead", null,
                React.createElement("tr", null,
                    React.createElement("th", null, "Quantity"),
                    React.createElement("th", null, "Item Description"),
                    React.createElement("th", null, "Price"))),
            orderedItem.map(function (cartItem, idx) { return React.createElement(CheckoutItem_1.CheckoutItem, { key: idx, itemprice: cartItem.price, attemptquantity: cartItem.quan, itemdescription: cartItem.name }); }),
            React.createElement("tbody", null,
                React.createElement("tr", null,
                    React.createElement("th", null),
                    React.createElement("th", null, "Price to Pay"),
                    React.createElement("th", null, orderedItem.reduce(reducer, 0))))),
        React.createElement("table", { className: "table w-full" },
            React.createElement(CheckoutInfo_1.CheckoutInfo, { key: idx, eachInfo: orderedDeliveryInfo[0] }))));
};
exports.DetailedTracking = DetailedTracking;
