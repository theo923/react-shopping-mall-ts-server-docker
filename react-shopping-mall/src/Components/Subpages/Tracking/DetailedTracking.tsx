import * as React from 'react'
import { CheckoutInfo } from '../ShoppingCart/Checkout/CheckoutInfo'
import { CheckoutItem } from '../ShoppingCart/Checkout/CheckoutItem'

interface ITrackingProps {
    order: any;
    idx: number;
}

export const DetailedTracking : React.FC<ITrackingProps> = ({ order, idx }) => {
    const reducer = (accumulator : number, currentObject: any) => accumulator + (currentObject.quan * currentObject.price);
    const orderedItem = order[0]
    const orderedDeliveryInfo = order[1]
    return(
        <div>
            <div>
                <div className="container glass text-neutral-content mb-6">
                    <figure className="px-6 relative">
                        <div style={{width:'266.67px', height:'150px'}} className="float-right">
                            <img src={orderedItem[0].url} alt={orderedItem[0].name} />
                        </div>
                    </figure> 
                    <div className="card-body max-h-full max-w-full relative">
                        <h2 className="card-title">Order #{idx}</h2> 
                        <div>{orderedItem.map((item: any, idx: number) => <div key={idx} className='italic'>{item.name} ${item.quan * item.price} @ {item.quan} </div>)}</div> 
                    </div>
                </div>
            </div>
            <table className="table w-full">
                    <thead>
                        <tr>
                            <th>Quantity</th> 
                            <th>Item Description</th> 
                            <th>Price</th>
                        </tr>
                    </thead> 
                {orderedItem.map((cartItem: any, idx: number) => <CheckoutItem key={idx} itemprice={cartItem.price} attemptquantity={cartItem.quan} itemdescription={cartItem.name} />)}
                    <tbody>
                        <tr>
                            <th></th> 
                            <th>Price to Pay</th> 
                            <th>{orderedItem.reduce(reducer, 0)}</th>
                        </tr>
                    </tbody> 
            </table>

            <table className="table w-full">
                <CheckoutInfo key={idx} eachInfo={orderedDeliveryInfo[0]} />
            </table>
        </div>

        
    )
}