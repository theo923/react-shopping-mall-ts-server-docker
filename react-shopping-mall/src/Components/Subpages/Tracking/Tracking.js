"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
exports.__esModule = true;
var React = require("react");
var TrackingCard_1 = require("./TrackingCard");
var DetailedTracking_1 = require("./DetailedTracking");
var Tracking = /** @class */ (function (_super) {
    __extends(Tracking, _super);
    function Tracking(props) {
        var _this = _super.call(this, props) || this;
        _this.handleViewDetail = function (ordernumber) {
            _this.setState({ orderNum: ordernumber, orderView: true });
        };
        _this.state = {
            orderView: false,
            orderNum: 0,
            orderList: []
        };
        return _this;
    }
    Tracking.prototype.componentDidMount = function () {
        var _this = this;
        this.readTrackApi()
            .then(function (data) {
            var selectedList = data.slice(0).reverse();
            _this.setState({ orderList: selectedList });
            console.log('render');
        });
    };
    Tracking.prototype.readTrackApi = function () {
        return __awaiter(this, void 0, void 0, function () {
            var response, data;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, fetch("http://" + ((process.env.REACT_APP_API_IP_ADDRESS === '') ? 'localhost' : process.env.REACT_APP_API_IP_ADDRESS) + ":" + process.env.REACT_APP_API_PORT + "/track", {
                            method: 'post',
                            headers: { 'Content-Type': 'application/json' },
                            body: JSON.stringify({
                                username: this.props.username
                            })
                        })];
                    case 1:
                        response = _a.sent();
                        data = response.json();
                        return [2 /*return*/, data];
                }
            });
        });
    };
    Tracking.prototype.render = function () {
        var _this = this;
        return (React.createElement("div", null, !this.state.orderView ?
            React.createElement("div", null,
                React.createElement("div", { className: 'text-4xl mb-6' },
                    "Tracking for ",
                    this.props.username),
                this.state.orderList.map(function (order, idx, arr) { return React.createElement(TrackingCard_1.TrackingCard, { key: idx, idx: arr.length - idx - 1, order: order, handleViewDetail: _this.handleViewDetail }); }))
            :
                React.createElement("div", null, this.state.orderList.map(function (order, idx, arr) { return (arr.length - idx - 1 === _this.state.orderNum) ? React.createElement(DetailedTracking_1.DetailedTracking, { key: idx, idx: arr.length - idx - 1, order: order }) : null; }))));
    };
    return Tracking;
}(React.Component));
exports["default"] = Tracking;
