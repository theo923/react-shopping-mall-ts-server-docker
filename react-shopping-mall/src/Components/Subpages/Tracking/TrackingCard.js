"use strict";
exports.__esModule = true;
exports.TrackingCard = void 0;
var React = require("react");
var TrackingCard = function (_a) {
    var order = _a.order, idx = _a.idx, handleViewDetail = _a.handleViewDetail;
    var orderedItem = order[0];
    return (React.createElement("div", null,
        React.createElement("div", null,
            React.createElement("div", { className: "card glass lg:card-side text-neutral-content mb-6" },
                React.createElement("figure", { className: "p-6 lg:py-0 mb-8" },
                    React.createElement("div", { style: { width: '266.67px', height: '150px' } },
                        React.createElement("img", { src: orderedItem[0].url, alt: orderedItem[0].name }))),
                React.createElement("div", { className: "card-body max-h-full max-w-full" },
                    React.createElement("h2", { className: "card-title" },
                        "Order #",
                        idx),
                    React.createElement("div", null, orderedItem.map(function (item, idx) { return React.createElement("div", { key: idx, className: 'italic' },
                        item.name,
                        " $",
                        item.quan * item.price,
                        " @ ",
                        item.quan,
                        " "); })),
                    React.createElement("div", { className: 'justify-end space-x-2 card-actions' },
                        React.createElement("button", { onClick: function () { return handleViewDetail(idx); }, className: "btn glass rounded-full" }, "Click for detailed info")))))));
};
exports.TrackingCard = TrackingCard;
