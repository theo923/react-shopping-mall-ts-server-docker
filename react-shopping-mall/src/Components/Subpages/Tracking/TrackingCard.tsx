import * as React from 'react'

interface ITrackingCardProps {
    order: any;
    idx: number;
    handleViewDetail(ordernumber: number): void;
}

export const TrackingCard : React.FC<ITrackingCardProps> = ({ order, idx, handleViewDetail }) => {
    const orderedItem = order[0]

    return(
            <div >
                <div>
                    <div className="card glass lg:card-side text-neutral-content mb-6">
                        <figure className="p-6 lg:py-0 mb-8">
                            <div style={{width:'266.67px', height:'150px'}}>
                                <img src={orderedItem[0].url} alt={orderedItem[0].name}/>
                            </div>
                        </figure> 
                        <div className="card-body max-h-full max-w-full">
                            <h2 className="card-title">Order #{idx}</h2> 
                            <div>{orderedItem.map((item: any, idx: number) => <div key={idx} className='italic'>{item.name} ${item.quan * item.price} @ {item.quan} </div>)}</div> 
                            <div className='justify-end space-x-2 card-actions'>
                                    <button onClick={() => handleViewDetail(idx)} className="btn glass rounded-full" >Click for detailed info</button>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
    )
}