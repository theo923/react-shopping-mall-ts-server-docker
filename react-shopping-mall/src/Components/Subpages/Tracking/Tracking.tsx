import * as React from 'react'
import { TrackingCard } from './TrackingCard'
import { DetailedTracking } from './DetailedTracking'

interface IState {
    orderView: boolean;
    orderNum: number;
    orderList: string[];
}

interface IProps {
    username: string;
}

class Tracking extends React.Component <IProps, IState> {
    constructor(props: IProps) {
        super(props)
        this.state = {
            orderView: false,
            orderNum: 0,
            orderList: []
        }
    }

    public componentDidMount() : void {
        this.readTrackApi()
        .then(data =>{
            const selectedList = data.slice(0).reverse()
            this.setState({orderList: selectedList})
            console.log('render')
        })
    }

    private async readTrackApi() {
        const response = await fetch(`http://localhost:${process.env.REACT_APP_API_PORT}/track`, {
            method: 'post',
            headers: {'Content-Type': 'application/json'},
            body: JSON.stringify({
                username: this.props.username
            })
        })
        const data = response.json()
        return data
    }

    private handleViewDetail = (ordernumber: number) : void => {
        this.setState({orderNum: ordernumber, orderView: true})
    }

    render() : JSX.Element {
        
        return(
            <div >
                { !this.state.orderView ? 
                <div>
                    <div className='text-4xl mb-6'>Tracking for {this.props.username}</div>
                    {this.state.orderList.map((order, idx, arr) => <TrackingCard key={idx} idx={arr.length - idx - 1} order={order} handleViewDetail={this.handleViewDetail} /> )}
                </div>
                :
                <div>
                {this.state.orderList.map((order, idx, arr) => (arr.length - idx - 1 === this.state.orderNum) ? <DetailedTracking key={idx} idx={arr.length - idx - 1} order={order} /> : null)}
                </div>
                }
            </div>
        )
    }

}

export default Tracking