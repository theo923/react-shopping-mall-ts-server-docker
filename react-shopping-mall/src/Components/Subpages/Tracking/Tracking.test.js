import React from 'react'
import Tracking from './Tracking';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

it('test for Tracking', () => {
    expect(shallow(<Tracking />)).toMatchSnapshot();
});