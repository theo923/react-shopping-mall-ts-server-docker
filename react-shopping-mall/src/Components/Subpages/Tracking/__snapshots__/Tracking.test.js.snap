// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Tracking 1`] = `ShallowWrapper {}`;
